### ISSUE TITLE
A better Flux overview for non-Flux users

### ISSUE THREAD

**Issue by batate** [FluxML/Flux.jl#1576]
Is there any desire for a pass through the documentation that: 

- Starts with a wholistic look at the Flux process: shape-training-data/build-model/train-model. (This is missing right now. The current flow takes the reader right to gradient descent.)
- Discusses gradient descent in terms of model params (weights and biases) rather than an arbitrary x
- Works through 2 examples (a line and then an xor gate) that help a user through a simple problem, and then a multi-dimensional one? 

I'm happy to put together three requests if this work would be interesting.

**Comment by DhairyaLGandhi** [comment]
Absolutely! I think that can be an interesting take on the flow of the docs and can act as a perspective guide in addition to the existing docs.

**Comment by Marc-Cox-08** [comment]
Yes agree Absolutely! an overview describing a workflow complete 
end to end thread from A to Z using Flux would definitely help.

In that regard here's a question and answer about 
stacking ensembles using Julia MLJ.jl that I feel
the documentation selected by Sebastian illustrates 
and overview well here >> 
	[https://alan-turing-institute.github.io/DataScienceTutorials.jl/getting-started/learning-networks/index.html](url)

--- Here is the whole Q&A excerpt ---

ML via Julia MLJ 
https://youtu.be/qSWbCn170HU 
And for Kaggle competitions the leader board positions 
are won by teams using ensembles aka MLJ Stacking 
as in MLJ allows Ensemble Stacking !

MLJ ensemble Stacking is great ! 
.. Now my Q: 
Q: In the MLJ workflow pipeline is there an automatic inverse function at the "end" of the pipeline to undo the function call at the "beginning" i.e. exp(log(x)) or equivalently f^-1(f(x)) = x 

Sebastian Vollmer
there is ` inverse_transform ` which does what you think; maybe have a look here: 
https://alan-turing-institute.github.io/DataScienceTutorials.jl/getting-started/learning-networks/index.html and see whether that answers your question ? 

Yes! That helps !

**Comment by batate** [comment]
Great. I will start work on something this weekend. I will initially add a guide/tutorial for raw beginners to get some early comments.

**Comment by darsnack** [comment]
Thanks! Ideally, our docs should start moving towards a tutorials + how-to's + internals + API. I think this kind of contribution will help get us there.

**Comment by batate** [comment]
darsnack, I agree. I would add overview. Getting into Flux, the big picture is what I missed the most. Here's the overview I want to write: 

Here's how you work with Flux:
- Shape training/test datasets
- build a model
- train
-- loss
-- optimizer
-- params
- verify

Now that the reader has context: Here's the secret sauce:
- autodiff: gradients   (in terms of W and b, not x)
- math/pure julia: examples


I have taken several passes through this with the Groxio content and I think it could be effective.

**Comment by batate** [comment]
Here's the first tutorial I want to write: 

Drill down: Build a predictor for an XOR gate: 

- Shape training/test datasets
-- describe the shape of multidimensional data inputs and outputs
- build a model
-- describe the design of dense layers (congratulations; you just built dense)
-- describe the arguments and parameters dimensions in dense layers
-- describe params of multilayer, multidimensional dense layers
- train 
-- build loss function from scratch
-- choose another optimizer, briefly describe tradeoffs
-- train! with a callback

Basically, this is a layered approach to documentation that lets the user skip big chunks.

**Comment by batate** [comment]
Finally, a quick start, that skips the above overview, linked from the top of the slow overview.

For the impatient: 

- An example that calls out 
-- build training and test data
-- build a multilayer model
-- train the model using APIs
-- Verify the trained model

(or whatever documentation seems appropriate)

**Comment by DhairyaLGandhi** [comment]
Maybe we could modify some of the model zoo examples, and have the text there. We have the script -> notebook setup too, so we can use that to deply to docs/ website.

**Comment by darsnack** [comment]
Can the model zoo deploy directly to https://fluxml.ai/Flux.jl/stable/ ?

**Comment by ToucheSir** [comment]
@batate have you seen https://fluxml.ai/tutorials/2020/09/15/deep-learning-flux.html (source [here](https://github.com/FluxML/model-zoo/tree/master/tutorials/60-minute-blitz))? It seems to fit in well with your quick start example.

**Comment by batate** [comment]
It does, doesn't it? A quick start would cut out the Julia parts, and cut out lots of the meta-description as well, and let the examples do most of the talking. 

I bet we could work with that.

**Comment by CarloLucibello** [comment]
Closing as addressed by the docs restructure. There is now an overview page ("Fitting a straight line"), a quickstart that walks through data / model / training loop end to end, and worked tutorials for linear regression, logistic regression and custom layers. Thanks for the push in this direction.
